Week3/homework/step2-6.js

```diff
@@ -1,16 +1,44 @@
 'use strict';
 
-const arr2d = [[1, 2], [3, 4], [5, 6]];
-const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
+const arr2d = [
+  [1, 2],
+  [3, 4],
+  [5, 6]
+];
+const arr3d = [
+  [
+    [1, 2],
+    [3, 4]
+  ],
+  [
+    [5, 6],
+    [7, 8]
+  ]
+];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  let flattenArr2 = [];
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      flattenArr2.push(arr[i][j])
+    }
+  }
+  return flattenArr2;
 }
+console.log(flattenArray2d(arr2d));
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const flattenArr3 = [];
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      for (let l = 0; l < arr[i][j].length; l++) {
+
+        flattenArr3.push(arr[i][j][l]);
+      }
+    }
+
+  }
+  return flattenArr3;
```

Comment:
This works, but you can also use the `.flat()` method :)

Reply:
@grod220 yeah , and it's much easier with .flat().